productserver/main.go

```
@@ -1,37 +1,82 @@
 package main
 
 import (
+	"database/sql"
 	"encoding/json"
+	"flag"
+	"fmt"
 	"log"
 	"net/http"
-	"os"
 	"strconv"
 
 	"github.com/gorilla/mux"
+	_ "github.com/mattn/go-sqlite3"
 )
 
 type Product struct {
-	ID          int    `json:"id"`
-	Name        string `json:"name"`
-	Description string `json:"description"`
-	Price       int    `json:"price"`
+	ID    int    `json:"id"`
+	Name  string `json:"name"`
+	Price int    `json:"price"`
 }
 
-var Products []Product
+var db *sql.DB
 
-func InitProduct() {
-	data, err := os.ReadFile("products.json")
+func InitProducts(path string) *sql.DB {
+	db, err := sql.Open("sqlite3", path)
 	if err != nil {
 		log.Fatal(err)
```

Better to return this error and let `main` decide if Fatal is the right way to go

---

Still need to do this